lib/OpenQA/Scheduler.pm

```diff
@@ -331,19 +335,29 @@ sub query_jobs {
                 'jobs_assets.asset_id' => $args{assetid},
             });
     }
-    if ($args{groupid}) {
-        push(
-            @conds,
-            {
-                'me.group_id' => $args{groupid},
-            });
+    if (defined $args{groupid}) {
+        if ($args{groupid} eq '0') {
+            push(
+                @conds,
+                {
+                    'me.group_id' => undef,
+                });
+        }
+        else {
+            push(
+                @conds,
+                {
+                    'me.group_id' => $args{groupid},
+                });
+        }
     }
```

Comment:
I think this whole construct can be written shorter as 

'me.group_id' => $args{groupid} || undef;